src/signature/mod.rs

```diff
@@ -39,6 +89,16 @@ pub fn verify_signature(
     }
 }
 
+/// Verify an ed25519 signature
+pub fn ed_verify(signature: &[u8], payload: &[u8], pubkey: &[u8]) -> bool {
+    let signature = match Ed25519Sig::from_bytes(signature) {
```

Comment:
It's good to return errors to the caller so the return type would be smth like: `Result<bool, &str>`
Ok(true) no error and sig ok
Ok(false) no error sig bad
Err() error happened, sig unknown